webpack/assets/javascripts/react_app/components/common/Slot/Slot.js

```diff
@@ -0,0 +1,44 @@
+import { cloneElement, isValidElement } from 'react';
+import PropTypes from 'prop-types';
+import { selectMaxComponent, selectComponentByWeight } from './SlotSelectors';
+
+const Slot = ({ fills, id, multi, children = null, ...props }) => {
+  const addProps = object => {
+    if (multi && !object.key) {
+      // eslint-disable-next-line no-console
+      console.warn('Please add a key attriubte to multiple fills');
+    }
+
+    if (isValidElement(object)) {
+      return cloneElement(object, { ...props });
+    }
+
+    if (!children) {
+      throw new Error('Slot with override props must have a child');
+    }
+
+    return cloneElement(children, { ...props, ...object });
+  };
+
+  if (Object.keys(fills).length) {
```

Comment:
Can we extract this logic to the "selectors" so this component will receive the `end-result` of those selectors as a prop?